plugin/validator_test.go

```diff
@@ -78,6 +78,50 @@ func TestValidator_LoadSchema(t *testing.T) {
 	})
 }
 
+func TestValidator_ValidatePath(t *testing.T) {
+	v, err := NewValidator(ValidatorOpts{})
+	assert.Nil(t, err)
+	schema, err := ioutil.ReadFile("testdata/path-test.lua")
+	assert.Nil(t, err)
+	pluginName, err := v.LoadSchema(string(schema))
+	assert.EqualValues(t, "path-test", pluginName)
+	assert.Nil(t, err)
+
+	plugin := `{
+	  "name": "path-test",
+	  "config": {
+		"path": "%s"
+	  },
+	  "enabled": true,
+	  "protocols": ["http"]
+	}`
+
+	goodPaths := []string{
+		"/",
+		"/path",
+		"/path/200",
+		"/path/200/",
+		"/path/200/test",
+		"/path/200/test/",
+		"/path/200/test-_~:@!$&'()*+,:=/",
+	}
+	for _, path := range goodPaths {
+		err := v.Validate(fmt.Sprintf(plugin, path))
+		assert.Nil(t, err)
+	}
+
+	badPaths := []string{
+		"path",
+		"path/200",
+		"/path/200?",
+		"/path/200#",
+	}
+	for _, path := range badPaths {
+		err := v.Validate(fmt.Sprintf(plugin, path))
+		assert.NotNil(t, err)
+	}
+}
+
```

Comment:
For completeness; could you add tests that reproduce [these from Kong Gateway busted tests](https://github.com/Kong/kong/blob/master/spec/01-unit/01-db/01-schema/05-services_spec.lua#L142-L248)